assets/src/stories-editor/helpers/index.js

```diff
@@ -1553,28 +1553,30 @@ export const getUniqueId = () => {
 /**
  * Returns an image of the first frame of a given video.
  *
- * @todo Perhaps allow specifying wanted image type.
- *
  * @param {string} src Video src URL.
  * @return {Promise<string>} The extracted image in base64-encoded format.
  */
 export const getFirstFrameOfVideo = ( src ) => {
 	const video = document.createElement( 'video' );
 	video.muted = true;
 	video.crossOrigin = 'anonymous';
+	video.preload = 'metadata';
+
+	return new Promise( ( resolve, reject ) => {
+		video.addEventListener( 'error', reject );
 
-	return new Promise( ( resolve ) => {
-		video.src = src;
-		video.addEventListener( 'loadeddata', () => {
+		video.addEventListener( 'canplay', () => {
 			const canvas = document.createElement( 'canvas' );
 			canvas.width = video.videoWidth;
 			canvas.height = video.videoHeight;
 
 			const ctx = canvas.getContext( '2d' );
 			ctx.drawImage( video, 0, 0, canvas.width, canvas.height );
 
-			canvas.toBlob( resolve );
+			canvas.toBlob( resolve, 'image/jpeg' );
 		} );
+
+		video.src = `${ src }#t=0.5`;
```

Comment:
Why the `#t=0.5`? If intending to advance to half a second into the video, I believe you rather need to use `video.src.currentTime = 0.5`.

Reply:
This is to seek forward a little bit so the poster image isn't just a black rectangle (because we're not playing the video). But it didn't occur to me that I can just use `video.currentTime` (not `video.src.currentTime`)

Reply:
Yes, `video.src.currentTime` was a typo. `video.currentTime` is what I meant.